ykbh/src/lib.rs

```diff
@@ -39,8 +39,8 @@ impl SIRInterpreter {
         }
     }
 
-    /// Sets the current frame of the interpreter.
-    pub fn set_frame(&mut self, tio: *mut usize) {
+    /// Inserts a pointer to the trace inputs into `locals`.
+    pub fn set_trace_inputs(&mut self, tio: *mut usize) {
```

Comment:
I *think* `tio` might better be `*mut u8` to be clearer to readers that it's just a big chunk of bytes. There's not *much* difference between `*mut u8` and `*mut usize`, but the former is more conventional for "random stuff".

Reply:
I used `*mut usize` to allude to the fact, that `tio` is a pointer. It also allows me to use `write::<usize>` below, which is little nicer than `copy`. But I can also change this to `*mut u8` if you like, there's not a big difference.

Reply:
`*mut T` says "this is is a pointer to zero or more `T`'s". `*mut usize` therefore says "I'm pointing to zero or more `usize`s", which isn't the case here. We really want to say "I'm pointing to zero more bytes".